src/mongo/MongoTripodConfig.class.php

```diff
@@ -117,189 +140,267 @@ private function loadConfig(Array $config)
+        $this->queueConfig["data_source"] = $this->getMandatoryKey("data_source",$queueConfig,'queue');
+        if(!isset($this->dataSources[$this->queueConfig['data_source']]))
         {
-            $this->ifCountExistsWithoutTTLThrowException($spec);
-            $this->viewSpecs[$spec[_ID_KEY]] = $spec;
+            throw new MongoTripodConfigException("Queue data source, " . $this->queueConfig['data_source'] . ", was not defined");
         }
 
-        // Load table specs
-        $tableSpecs = (array_key_exists("table_specifications",$config)) ? $config["table_specifications"] : array();
-        foreach ($tableSpecs as $spec)
+        // A 'pod' corresponds to a logical database
+        $this->databases = $this->getMandatoryKey("stores",$config);
+        $defaultDB = null;
+        foreach ($this->databases as $storeName=>$storeConfig)
         {
-            if(!isset($spec[_ID_KEY]))
+            $this->dbConfig[$storeName] = array ("data_source"=>$this->getMandatoryKey("data_source",$storeConfig));
+            if(isset($storeConfig['database']) && !empty($storeConfig['database']))
             {
-                throw new MongoTripodConfigException("Table spec does not contain " . _ID_KEY);
+                $this->dbConfig[$storeName]["database"]=$storeConfig['database'];
+            }
+            else
+            {
+                $this->dbConfig[$storeName]["database"] = $storeName;
             }
 
-            // Get all "fields" in the spec
-            $fieldsInTableSpec = $this->findFieldsInTableSpec('fields', $spec);
-            // Loop through fields and validate
-            foreach($fieldsInTableSpec as $field)
+            $this->cardinality[$storeName] = array();
+            $this->indexes[$storeName] = array();
+            $this->podConnections[$storeName] = array();
+            if(isset($storeConfig["pods"]))
             {
-                if (!isset($field['fieldName']))
+                foreach($storeConfig["pods"] as $podName=>$podConfig)
                 {
-                    throw new MongoTripodConfigException("Field spec does not contain fieldName");
-                }
+                    $dataSource = (isset($podConfig['data_source']) ? $podConfig['data_source'] : $storeConfig['data_source']);
+                    $this->podConnections[$storeName][$podName] = $dataSource;
 
-                if(isset($field['predicates']))
-                {
-                    foreach($field['predicates'] as $p)
+                    // Set cardinality, also checking against defined namespaces
+                    if (array_key_exists('cardinality', $podConfig))
+                    {
+                        // Test that the namespace exists for each cardinality rule defined
+                        $cardinality = $podConfig['cardinality'];
+                        foreach ($cardinality as $qname=>$cardinalityValue)
+                        {
+                            $namespaces = explode(':', $qname);
```

Comment:
This would be a breaking change to the config.  It also wasn't introduced in this work.